src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1110,8 +1112,15 @@ private void onDecodeUnsubscribe(
                     onDecodeError(traceId, authorization, PROTOCOL_ERROR);
                     return;
                 }
```

Comment:
Suggest tracking the decode error in the loop and breaking the loop on problem.
Then after loop, check for error and call `onDecodeError`, otherwise call `doEncodeUnsubAck`.